src/Composer/ComposerWidget.vala

```diff
@@ -608,14 +639,61 @@ public class Mail.ComposerWidget : Gtk.Grid {
         html.construct_from_stream_sync (stream_filter);
         html.set_mime_type ("text/html; charset=utf-8");
 
-        var body = new Camel.Multipart ();
-        body.set_mime_type ("multipart/alternative");
-        body.set_boundary (null);
-
         var part = new Camel.MimePart ();
         part.content = html;
         part.set_encoding (Camel.TransferEncoding.ENCODING_QUOTEDPRINTABLE);
+
+        var body = new Camel.Multipart ();
+        body.set_mime_type ("multipart/alternative");
+        body.set_boundary (null);
         body.add_part (part);
+
+        if (attachment_box.get_children ().length () > 0) {
+            foreach (unowned Gtk.Widget attachment in attachment_box.get_children ()) {
+                if (attachment is Attachment) {
```

Comment:
```suggestion
                if (!(attachment is Attachment)) {
                    continue;
                }
```
This would allow to de-indent it a bit